src/main/java/cam72cam/immersiverailroading/model/part/PartSound.java

```diff
@@ -31,36 +49,73 @@ public void effects(EntityMoveableRollingStock stock, float volume) {
+            ISound endSound = soundTrio.end; //Unused for now
 
             if (volume > 0) {
-                sound.setPosition(stock.getPosition());
-                sound.setVelocity(stock.getVelocity());
-                sound.setVolume(volume);
-                sound.setPitch(pitch);
-
-                if (!sound.isPlaying()) {
-                    sound.play(stock.getPosition());
-                } else {
-                    sound.update();
+                if(startSound != null) {
+                    startSound.setPosition(stock.getPosition());
+                    startSound.setVelocity(stock.getVelocity());
+                    startSound.setVolume(volume);
+                    startSound.setPitch(pitch);
+
+                    if (!startSound.isPlaying() && startIsPlaying) { //Start sound is done playing one take
+                        startHasPlayed = true;
+                        startIsPlaying = false;
+                    } else if(!startSound.isPlaying() && !startHasPlayed) { //If nothing has started yet
+                        startSound.play(stock.getPosition());
+                        startIsPlaying = true;
+                    } else {
+                        startSound.update();
+                    }
+                }
+                if(startHasPlayed || startSound == null) { //Once start has played, move on to loop
+                    loopSound.setPosition(stock.getPosition());
+                    loopSound.setVelocity(stock.getVelocity());
+                    loopSound.setVolume(volume);
+                    loopSound.setPitch(pitch);
+
+                    if(!loopSound.isPlaying())
+                    {
+                        loopSound.play(stock.getPosition());
+                    } else {
+                        loopSound.update();
+                    }
                 }
             } else {
-                if (sound.isPlaying()) {
-                    sound.stop();
+                if (startSound != null && startSound.isPlaying()) {
+                    startSound.stop();
+                }
+                if (loopSound != null && loopSound.isPlaying()) {
+                    loopSound.stop();
+                }
+                if (endSound != null && endSound.isPlaying()) {
+                    endSound.stop();
                 }
+                startHasPlayed = false;
+                endHasPlayed = false; //Probably needs to be somewhere else when endSound has usage
             }
         }
     }
 
     public void removed(EntityMoveableRollingStock stock) {
-        ISound sound = sounds.get(stock.getUUID());
-        if (sound != null) {
-            sound.terminate();
+        Sounds soundTrio = sounds.get(stock.getUUID());
+        if (soundTrio != null) {
+            if(soundTrio.start != null){ soundTrio.start.terminate();}
+            if(soundTrio.loop != null){ soundTrio.loop.terminate();}
+            if(soundTrio.end != null){ soundTrio.end.terminate();}
```

Comment:
sounds.remove(stock.getUUID())

Reply:
Took a second look and this is not something you introduced, but is a bug in the original code.

Reply:
Thanks for pointing out that it's an ExpireableMap and that's not needed!